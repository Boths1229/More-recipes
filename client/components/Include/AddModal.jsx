import React from 'react';
import PropTypes, { any } from 'prop-types';

/**
 * @description functional component to render modal for adding recipes
 *
 * @param {props} props
 *
 * @returns {JSX} return JSX
 */
const AddModal = (props) => {
  const {
    onChange, value, saveImageToCloud, addRecipe
  } = props;
  const { name, description, ingredient } = value;
  return (
    <div
      className="modal fade"
      id="exampleModal"
      tabIndex="-1"
      role="dialog"
      aria-labelledby="exampleModalLabel"
      aria-hidden="true"
    >
      <div className="modal-dialog" role="document">
        <div className="modal-content">
          <div className="modal-header black">
            <h5 className="modal-title black" id="exampleModalLabel">
              Add new recipe
            </h5>
            <button
              type="button"
              className="close"
              data-dismiss="modal"
              aria-label="Close"
            >
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
          <div className="modal-body">
            <form>
              <div className="form-group">
                <label
                  htmlFor="recipient-name"
                  className="col-form-label black"
                >
                  Name &#58;
                </label>
                <input
                  name="name"
                  onChange={onChange}
                  value={name}
                  type="text"
                  className="form-control"
                  id="recipient-name"
                />
              </div>
              <div className="form-group">
                <label htmlFor="message-text" className="col-form-label black">
                  Description &#58;
                </label>
                <textarea
                  name="description"
                  onChange={onChange}
                  value={description}
                  className="form-control"
                  id="message-text"
                />
              </div>
              <div className="form-group">
                <label htmlFor="ingredient" className="col-form-label black">
                  Ingredient &#58;
                </label>
                <input
                  id="ingredient"
                  type="text"
                  name="ingredient"
                  onChange={onChange}
                  value={ingredient}
                  className="form-control"
                />
              </div>
              <div className="form-group">
                <input
                  type="file"
                  name="image"
                  accept="image/*"
                  onChange={saveImageToCloud}
                />
              </div>
            </form>
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="btn btn-secondary"
              data-dismiss="modal"
            >
              Close
            </button>
            <button
              type="button"
              className="btn btn-orange"
              onClick={addRecipe}
              data-dismiss="modal"
            >
              Add recipe
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

AddModal.propTypes = {
  addRecipe: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.objectOf(any).isRequired
};


export default AddModal;
