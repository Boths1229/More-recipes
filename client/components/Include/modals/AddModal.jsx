import React from 'react';
import PropTypes, { any } from 'prop-types';

/**
 * @description functional component to render modal for adding recipes
 *
 * @param {props} props
 *
 * @returns {JSX} return JSX
 */
const AddModal = (props) => {
  const {
    onChange, value, saveImageToCloud, addRecipe, recipeImage
  } = props;
  const {
    name, description, ingredient, responseMessage
  } = value;
  return (
    <div
      className="modal fade"
      id="exampleModal"
      tabIndex="-1"
      role="dialog"
      aria-labelledby="exampleModalLabel"
      aria-hidden="true"
    >
      <div className="modal-dialog" role="document">
        <div className="modal-content">
          <div className="modal-header black">
            <h5 className="modal-title black" id="exampleModalLabel">
              Add new recipe
            </h5>
            <button
              type="button"
              className="close"
              data-dismiss="modal"
              aria-label="Close"
            >
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
          <div className="modal-body">
            <form>
              <div className="form-group">
                <div className="error-message">
                  { responseMessage }
                </div>
                <label
                  htmlFor="recipient-name"
                  className="col-form-label black"
                >
                  Name &#58;
                </label>
                <input
                  name="name"
                  onChange={onChange}
                  value={name}
                  type="text"
                  className="form-control"
                  id="recipient-name"
                />
              </div>
              <div className="form-group">
                <label htmlFor="message-text" className="col-form-label black">
                  Description &#58;
                </label>
                <textarea
                  name="description"
                  onChange={onChange}
                  value={description}
                  className="form-control"
                  id="message-text"
                />
              </div>
              <div className="form-group">
                <label htmlFor="ingredient" className="col-form-label black">
                  Ingredient &#58;
                </label>
                <input
                  id="ingredient"
                  placeholder="separate each ingredient with a comma"
                  type="text"
                  name="ingredient"
                  onChange={onChange}
                  value={ingredient}
                  className="form-control"
                />
              </div>
              <div className="form-group img-text">
                <input
                  type="file"
                  name="image"
                  accept="image/*"
                  onChange={saveImageToCloud}
                />
              </div>
            </form>
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="btn btn-secondary"
              data-dismiss="modal"
            >
              Close
            </button>
            <button
              type="button"
              disabled={!recipeImage ? 'disabled' : null}
              className="btn btn-orange"
              onClick={addRecipe}
            >
              Add recipe
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

AddModal.propTypes = {
  addRecipe: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  saveImageToCloud: PropTypes.func.isRequired,
  value: PropTypes.objectOf(any).isRequired
};


export default AddModal;
