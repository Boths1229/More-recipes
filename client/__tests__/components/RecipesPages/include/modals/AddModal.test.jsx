import React from 'react';
import { shallow } from 'enzyme';
import AddModal from '../../../../../components/Include/modals/AddModal';

describe('Add modal component', () => {
  it('matches snapshot', () => {
    const props = {
      value: {}
    };
    const output = shallow(<AddModal {...props} />);
    expect(output).toMatchSnapshot();
  });
});
