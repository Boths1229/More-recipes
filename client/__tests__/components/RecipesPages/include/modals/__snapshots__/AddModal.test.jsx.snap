// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add modal component matches snapshot 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="exampleModalLabel"
  className="modal fade"
  id="exampleModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header black"
      >
        <h5
          className="modal-title black"
          id="exampleModalLabel"
        >
          Add new recipe
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <form>
          <div
            className="form-group"
          >
            <div
              className="error-message"
            />
            <label
              className="col-form-label black"
              htmlFor="recipient-name"
            >
              Name :
            </label>
            <input
              className="form-control"
              id="recipient-name"
              name="name"
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="message-text"
            >
              Directions :
            </label>
            <textarea
              className="form-control"
              id="message-text"
              name="description"
              placeholder="seperate each directions with a newline"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="ingredient"
            >
              Ingredient :
            </label>
            <input
              className="form-control"
              id="ingredient"
              name="ingredient"
              placeholder="separate each ingredient with a comma"
              type="text"
            />
          </div>
          <div
            className="form-group img-text"
          >
            <input
              accept="image/*"
              name="image"
              type="file"
            />
          </div>
        </form>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          type="button"
        >
          Close
        </button>
        <button
          className="btn btn-orange"
          disabled="disabled"
          type="button"
        >
          Add recipe
        </button>
      </div>
    </div>
  </div>
</div>
`;
